### PR TITLE
ENH: Improvement over the Xdawn Transformer example

### PR DESCRIPTION
**Pull request by alexandrebarachant**
This PR is an improvement of the Xdawn transformer example. It is made to replace the `plot_decoding_csp_space.py` example. Decoding accuracy is now 99.6% instead of 90.3% for the CSP example.

Using CSP for decoding of evoked potential (with log-variance as feature) is both conceptually wrong and inefficient in practice. I suggest we remove the CSP example. MNE should not showcase bad habits. the `plot_decoding_csp_eeg.py` example is already covering the correct way to use CSP (to decode induced activity).

### CODE REVIEW

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 1]
```diff
@@ -1,50 +1,77 @@
...
+#          Alexandre Barachant <alexandre.barachant@gmail.com>
 #
 # License: BSD (3-clause)
+import numpy as np
 
-from sklearn.cross_validation import cross_val_score
+from sklearn.cross_validation import cross_val_score, ShuffleSplit  # noqa
 from sklearn.pipeline import make_pipeline
-from sklearn.linear_model import LogisticRegression
+from sklearn.svm import SVC  # noqa
```
Comment (kingjr) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: why is the noqa necessary?

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 2]
```diff
@@ -1,50 +1,77 @@
...
 raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
 event_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
-tmin, tmax = -0.1, 0.3
+tmin, tmax = -0.2, 0.5
 event_id = dict(aud_l=1, vis_l=3)
 
 # Setup for reading the raw data
 raw = io.read_raw_fif(raw_fname, preload=True)
-raw.filter(1, 20, method='iir')
+raw.filter(2, None, method='iir')
```
Comment (kingjr) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: 2 => 2. to eplicitly use float

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 3]
```diff
@@ -1,50 +1,77 @@
...
 events = read_events(event_fname)
 
-picks = pick_types(raw.info, meg=False, eeg=True, stim=False, eog=False,
+picks = pick_types(raw.info, meg='grad', eeg=False, stim=False, eog=False,
```
Comment (kingjr) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: 'mag' will be faster for the example

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 4]
```diff
@@ -1,50 +1,77 @@
...
 y = label_binarize(epochs.events[:, 2], classes=[1, 3]).ravel()
 
-clf = make_pipeline(XdawnTransformer(n_components=3),
+clf = make_pipeline(XdawnTransformer(n_components=2),
                     Vectorizer(),
-                    LogisticRegression())
-score = cross_val_score(clf, X, y, cv=5)
-print(score)
+                    StandardScaler(),
+                    SVC(C=1, kernel='linear'))
```
Comment (kingjr) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: I would use a a logistic regression, it's faster, and no extra param

File: examples/decoding/plot_decoding_xdawn_transformer.py (hunk 3 above)
Comment (alexandrebarachant) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: it give nicer patterns too :)

**Comment by kingjr** [comment]
LGTM

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 5]
```diff
@@ -1,50 +1,77 @@
...
 event_id = dict(aud_l=1, vis_l=3)
 
 # Setup for reading the raw data
 raw = io.read_raw_fif(raw_fname, preload=True)
-raw.filter(1, 20, method='iir')
+raw.filter(2., None, method='iir')
 events = read_events(event_fname)
 
-picks = pick_types(raw.info, meg=False, eeg=True, stim=False, eog=False,
+picks = pick_types(raw.info, meg='mag', eeg=False, stim=False, eog=False,
```
Comment (agramfort) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: hehe you boost performance by using MEG :)

File: examples/decoding/plot_decoding_xdawn_transformer.py [hunk 6]
```diff
@@ -1,50 +1,77 @@
...
+from sklearn.preprocessing import StandardScaler
 
 from mne import io, pick_types, read_events, Epochs
+from mne.viz import plot_topomap
 from mne.datasets import sample
 from mne.decoding import Vectorizer
 from mne.preprocessing import XdawnTransformer
 
-data_path = sample.data_path()
+import matplotlib.pyplot as plt
```
Comment (agramfort) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: import matplotlib just after numpy.

File: examples/decoding/plot_decoding_xdawn_transformer.py (hunk 5 above)
Comment (alexandrebarachant) [inline examples/decoding/plot_decoding_xdawn_transformer.py]: This is the same for the CSP example (but using 'mag' instead of 'grad'). All type of sensor give the same results (99.6%), including EEG.

**Comment by agramfort** [comment]
Thanks
